cvat-ui/src/actions/annotation-actions.ts

```diff
@@ -1608,3 +1611,84 @@ export function restoreFrameAsync(frame: number): ThunkAction {
         }
     };
 }
+
+export function updateEditedStateAsync(
+    shapeType: ShapeType | null,
+    editedStateInstance: ObjectState | null,
+): ThunkAction {
+    return async (dispatch: ThunkDispatch, getState): Promise<void> => {
+        if (editedStateInstance) {
+            const state = getState();
+            const { instance: canvas } = state.annotation.canvas;
+            if (canvas) {
+                (canvas as Canvas).configure({
+                    hideEditedObject: editedStateInstance.hidden,
+                });
+            }
+
+            dispatch({
+                type: AnnotationActionTypes.HIDE_EDITED_STATE,
+                payload: {
+                    hide: editedStateInstance.hidden,
+                },
+            });
+        }
+        dispatch({
+            type: AnnotationActionTypes.UPDATE_EDITED_STATE,
+            payload: {
+                shapeType,
+                editedStateInstance,
+            },
+        });
+    };
+}
+
+export function resetEditedStateAsync(): ThunkAction {
+    return async (dispatch: ThunkDispatch, getState): Promise<void> => {
+        const state = getState();
+        const { instance: canvas } = state.annotation.canvas;
+        const { editedStateHidden } = state.annotation.editing;
+        if (canvas && editedStateHidden) {
+            (canvas as Canvas).configure({
+                hideEditedObject: false,
+            });
+        }
+
+        dispatch({
+            type: AnnotationActionTypes.RESET_EDITED_STATE,
+            payload: {},
+        });
```

Comment:
`dispatch(resetEditedState())`?

Reply:
Not really relevant considering my other comment with proposal to combine actions